src/lib/autoLaunch/winAutoLaunch.ts

```diff
@@ -1,56 +1,29 @@
-/* eslint-disable no-unused-vars */
-
-// const regKey = new Winreg({
-//     hive: Winreg.HKCU,
-//     key:  '\\Software\\Microsoft\\Windows\\CurrentVersion\\Run'
-// });
-// path = require 'path'
-import path from 'path'
+import { AutoLaunchParams, ChildReturnData } from '../types'
 import * as native from '../native'
-import { AutoLaunchParams } from '../types'
-
-module.exports = {
-
-  enable({ appName, appPath, hidden }: AutoLaunchParams) {
-    // return new Promise(function (resolve, reject) {
-    //   let pathToAutoLaunchedApp = appPath;
-    //   let args = '';
-    //   const updateDotExe = path.join(path.dirname(process.execPath), '..', 'update.exe');
 
-    //   if (((process.versions != null ? process.versions.electron : undefined) != null) && native.dirExists(updateDotExe)) {
-    //     pathToAutoLaunchedApp = updateDotExe;
-    //     args = ` --processStart \"${path.basename(process.execPath)}\"`;
-    //     if (isHiddenOnLaunch) { args += ' --process-start-args "--hidden"'; }
-    //   } else {
-    //     if (isHiddenOnLaunch) { args += ' --hidden'; }
-    //   }
+const subKey = "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Run"
 
-    //   return regKey.set(appName, Winreg.REG_SZ, `\"${pathToAutoLaunchedApp}\"${args}`, function (err) {
-    //     if (err != null) { return reject(err); }
-    //     return resolve();
-    //   });
-    // });
+const winAutoLaunch = {
+  // eslint-disable-next-line no-unused-vars
+  async enable({ appName, appPath, hidden }: AutoLaunchParams): Promise<ChildReturnData> {
+    let returnVal = <ChildReturnData>{ stdout: [], stderr: [] }
+    const result = <string>(await native.winregSet(subKey, appName, appPath))
```

Comment:
`native.winregSet` already has return type specified, no need to repeat it, much easier to read this way
```suggestion
    const result = await native.winregSet(subKey, appName, appPath)
```